### PR TITLE
[runtime] [2020-02] Fix GC problems in create_cattr_named/typed.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/custom-attrs.c [hunk 1]
```diff
@@ -566,21 +569,27 @@ MONO_RESTORE_WARNING
...
-		case MONO_TYPE_SZARRAY:
+		case MONO_TYPE_SZARRAY: {
+			HANDLE_FUNCTION_ENTER ();
+			MONO_HANDLE_NEW (MonoArray, arr);
+
 			for (i = 0; i < alen; i++) {
 				MonoObject *item = NULL;
 				load_cattr_value (image, m_class_get_byval_arg (tklass), &item, p, boundp, &p, error);
 				if (!is_ok (error))
 					return NULL;
```
Comment (srxqds) [inline mono/metadata/custom-attrs.c:580]: should call HANDL_FUNCTION_RETURN() befor return?

Comment (jaykrell) [reply mono/metadata/custom-attrs.c:580]: It does look wrong, agreed, but your suggestion seems wrong too.
@vargaz Make it single-point-of-return, please. Well. Not exactly, but you understand.

**Comment by BrzVlad** [comment]
why do we need this ? Wasn't this impacting only wasm with gc enabled ?

**Comment by vargaz** [comment]
Guess so.

**Pull request closed without merging**